common/src/main/java/net/opentsdb/data/TimeSeriesId.java

```diff
@@ -0,0 +1,122 @@
+//This file is part of OpenTSDB.
+//Copyright (C) 2017  The OpenTSDB Authors.
+//
+//This program is free software: you can redistribute it and/or modify it
+//under the terms of the GNU Lesser General Public License as published by
+//the Free Software Foundation, either version 2.1 of the License, or (at your
+//option) any later version.  This program is distributed in the hope that it
+//will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+//of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser
+//General Public License for more details.  You should have received a copy
+//of the GNU Lesser General Public License along with this program.  If not,
+//see <http://www.gnu.org/licenses/>.
+package net.opentsdb.data;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An identifier for a time series. The identity can be as simple as the alias
+ * or a combination of namespace, metrics, tags, etc.
+ * <p>
+ * Most of the fields return a list or map. These represent the aggregate of all
+ * time series included in the sourcing of this series. This is applicable to
+ * synthetic series when multiple time series are aggregated into one series.
+ * <p>
+ * In general, if a raw time series is the source then only one metric and
+ * an optional set of tags should be present.
+ * 
+ * @since 3.0
+ */
+public interface TimeSeriesId extends Comparable<TimeSeriesId> {
+  /**
+   * @return True if the fields are encoded using a format specified by the 
+   * storage engine.
+   */
+  public boolean encoded();
+  
+  /**
+   * A simple string for identifying the time series. The alias may be null or
+   * empty. If a value is present, the alias must be unique within a set of
+   * time series.
+   * <p>
+   * Invariant: If all other fields for the identity are empty, then the alias
+   * must be non-null and non-empty.
+   * 
+   * @return A string or null if not used.
+   */
+  public String alias();
+  
+  /**
+   * A list of namespace identifiers included in the sources for this time series.
+   * May be empty if namespaces are not in use for the platform.
+   *  
+   * @return A non-null list if zero or more namespace identifiers.
+   */
+  public List<String> namespaces();
```

Comment:
What is a namespace? A link to documentation may help here.

Reply:
Will link. Used for multi-tenancy